infrastructure/network/utils.go

```diff
@@ -0,0 +1,64 @@
+package network
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/vishvananda/netlink"
+	"github.com/weaveworks/flintlock/core/models"
+)
+
+const (
+	ifaceLength       = 7
+	retryGenerate     = 5
+	randomBytesLength = 32
+	prefix            = "fl"
+	tapPrefix         = "tap"
+	// It's vtap only to save space.
+	macvtapPrefix = "vtap"
+)
+
+func NewIfaceName(ifaceType models.IfaceType) (string, error) {
+	devPrefix := ""
+
+	switch ifaceType {
+	case models.IfaceTypeTap:
+		devPrefix = fmt.Sprintf("%s%s", prefix, tapPrefix)
+	case models.IfaceTypeMacvtap:
+		devPrefix = fmt.Sprintf("%s%s", prefix, macvtapPrefix)
+	case models.IfaceTypeUnsupported:
+		return "", interfaceErrorf("unsupported interface type: %s", ifaceType)
+	default:
+		return "", interfaceErrorf("unknown interface type: %s", ifaceType)
+	}
+
+	for i := 0; i < retryGenerate; i++ {
+		name, err := generateRandomName(devPrefix)
+		if err != nil {
+			continue
+		}
```

Comment:
i don't think the retry generation bit (or the failure of it) is covered by the tests, although i grant you it would be a tricky one to test 😆  so maybe that's fine

Reply:
I rather not create a whole struct and mock it only to test when netlink throws an error. Same goes to the "what happened when io.ReadFull(rand.Reader, id) throws an error".

(I know it's not covered, my nvim tells me :wink: )

Reply:
😱 what plugin do you use for that?!? Here i am reading the code like a sucker 🤦‍♀️

Reply:
`vim-go` has a `:GoCoverageToggle` command.